connect/runtime/src/test/java/org/apache/kafka/connect/util/RetryUtilTest.java

```diff
@@ -116,16 +117,20 @@ public void testNoBackoffTimeAndSucceed() throws Exception {
                 .thenThrow(new TimeoutException())
                 .thenReturn("success");
 
-        assertEquals("success", RetryUtil.retryUntilTimeout(mockCallable, testMsg, TIMEOUT, 0));
+        assertEquals("success", RetryUtil.retryUntilTimeout(mockCallable, testMsg, Duration.ofMillis(100), 0, mockTime));
         Mockito.verify(mockCallable, Mockito.times(4)).call();
     }
 
     @Test
     public void testNoBackoffTimeAndFail() throws Exception {
-        Mockito.when(mockCallable.call()).thenThrow(new TimeoutException("timeout exception"));
+        Mockito.when(mockCallable.call()).thenAnswer(invocation -> {
+            // Without any backoff time, the speed of the operation itself limits the number of retries and retry rate.
+            mockTime.sleep(30);
+            throw new TimeoutException("timeout exception");
+        });
```

Comment:
We can use the auto-tick feature of the `MockTime` class instead of adding in fake calls to `sleep` from our mock object.
```suggestion
        // Without backoff time, we never sleep; auto-tick helps move things along
        mockTime = new MockTime(10);
        Mockito.when(mockCallable.call()).thenThrow(new TimeoutException("timeout exception"));
```

Reply:
oh cool I didn't understand the intent of that feature, it's perfect for this situation!